src/importer/index.js

```diff
@@ -175,41 +168,49 @@ function createAndStoreFile (file, ipldResolver, cb) {
       // create a parent node and add all the leafs
 
       const f = new UnixFS('file')
-      const n = new DAGNode()
-
-      for (let leaf of leaves) {
-        f.addBlockSize(leaf.leafSize)
-        n.addRawLink(
-          new DAGLink(leaf.Name, leaf.Size, leaf.Hash)
-        )
-      }
+      let n
 
-      n.data = f.marshal()
-
-      n.multihash((err, multihash) => {
-        if (err) {
-          return cb(err)
-        }
-
-        ipldResolver.put({
-          node: n,
-          cid: new CID(multihash)
-        }, (err) => {
-          if (err) {
-            return cb(err)
-          }
-
-          n.size((err, size) => {
+      series([
+        (cb) => {
+          DAGNode.create(f.marshal(), (err, node) => {
```

Comment:
It is here where I'm failing, I'm passing the f.marshal() before I do the addBlockSize